raiden-ts/src/transport/utils.ts

```diff
@@ -33,33 +34,52 @@ export const getPresences$: (action$: Observable<RaidenAction>) => Observable<Pr
 );
 
 /**
- * Stringify a caps mapping
+ * Stringify a caps mapping to a caps url
+ * E.g.'mxc://raiden.network/cap?k1=true&k2=v2&k2=v3&k4=null&k5=123'
  *
  * @param caps - Capabilities object/mapping
  * @returns stringified version of caps
  */
 export function stringifyCaps(caps: Caps): string {
-  return Object.entries(caps)
-    .filter(([, v]) => typeof v !== 'boolean' || v)
-    .map(([k, v]) => (typeof v === 'boolean' ? k : `${k}="${v}"`))
-    .join(',');
+  const url = new URL('mxc://raiden.network/cap');
+  // URLSearchParams.append can handle all primitives
+  const appendParam = (key: string, value: string | number | boolean | null) =>
+    url.searchParams.append(key, value as string);
+  for (const [key, value] of Object.entries(caps)) {
+    if (Array.isArray(value)) value.forEach(appendParam.bind(null, key));
+    else appendParam(key, value);
+  }
+  return url.href;
 }
 
 /**
- * Parse a caps string in the format 'k1,k2=v2,k3="v3"' to { k1: true, k2: v2, k3: v3 } object
+ * Parse a caps string in the format 'mxc://raiden.network/cap?k1=true&k2=v2&k2=v3&k4=null&k5=123'
+ * to a { k1: true, k2: ['v2','v3'], k4: null, k5: 123 } object
  *
  * @param caps - caps string
  * @returns Caps mapping object
  */
 export function parseCaps(caps?: string | null): Caps | undefined {
   if (!caps) return;
-  const result: { [k: string]: string | boolean } = {};
+  const result: Mutable<Caps> = {};
   try {
-    // this regex splits by comma, but respecting strings inside double-quotes
-    for (const cap of caps.split(/,(?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)/g)) {
-      const match = cap.match(/^\s*([^=]+)(?: ?= ?"?(.*?)"?\s*)?$/);
-      if (match) result[match[1]] = match[2] ?? true;
-    }
+    const url = new URL(caps!);
```

Comment:
Why is the force here necessary? Shouldn't TypeScript now that there is the `if` condition before? At least when I remove the `!` locally I don't get any error. (I verified that when I remove the if-line I get the error)

Reply:
Nice catch, yes, didn't notice it. Thanks!